### PR TITLE
feat: 🎸 Table chart column format

### PR DESCRIPTION
**Pull request by mczapkowicz**
Part of solution for: https://metakeen.atlassian.net/jira/software/projects/FEE/boards/10?selectedIssue=FEE-599

### CODE REVIEW

File: src/charts/table/transform.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { TableChartSettings } from '@keen.io/charts';
+import { WidgetTransform } from '../../types';
+
+const transform: WidgetTransform<TableChartSettings> = {
```
Comment (maciejrybaniec) [inline src/charts/table/transform.ts:4]: The `table` transformation should be added to `serializeInputSettings` function (switch - case block).

File: src/serializers/serializeOutputSettings.ts [hunk 2]
```diff
@@ -33,6 +33,8 @@ const serializeOutputSettings = (
       return chartTransformations.line.serializeOut(chartSettings);
     case 'bar':
       return chartTransformations.bar.serializeOut(chartSettings);
+    // case 'table':
```
Comment (maciejrybaniec) [inline src/serializers/serializeOutputSettings.ts:36]: This should be uncommented

File: src/types.ts [hunk 3]
```diff
@@ -50,12 +51,15 @@ export type WidgetTransform<S> = {
 };
 
 export type FormatterSettings = {
```
Comment (maciejrybaniec) [inline src/types.ts:53]: IMO we should set `FormatterSettings` as:
```
type FormatterSettings = NumericFormatter | DateFormatter | StringFormatter;
```

Right now everything is mixed and it's hard to know what to expect in particular use case.

**Review by maciejrybaniec — CHANGES_REQUESTED** [review]

File: public/locales/en/translation.json [hunk 4]
```diff
@@ -66,7 +66,7 @@
     "saved_query_name_tooltip": "Click to use as Chart Title"
   },
   "widget_customization_format_value_settings": {
-    "section_title": "Value Format",
+    "section_title": "Values Format",
```
Comment (apryka) [inline public/locales/en/translation.json:69]: according to https://metakeen.atlassian.net/browse/FEE-777 this should be `Value Format`

File: public/locales/en/translation.json [hunk 5]
```diff
@@ -77,8 +77,24 @@
     "select_pattern_placeholder": "Select pattern",
     "select_operations_placeholder": "Select",
     "original_data": "Original data",
+    "full_number": "(Full number)",
```
Comment (apryka) [inline public/locales/en/translation.json:80]: according to https://metakeen.atlassian.net/browse/FEE-777 this should be removed

File: src/components/ComponentSettings/components/Circular/components/ValueMode/ValueMode.tsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
 import React, { FC } from 'react';
 import { RadioSelect } from '@keen.io/ui-core';
 
-import Row from '../Row';
-import Label from '../Label';
+import Row from '../../../../../Row';
+import Label from '../../../../../Label';
```
Comment (apryka) [inline src/components/ComponentSettings/components/Circular/components/ValueMode/ValueMode.tsx:5]: pretty long path, what do you think about refactoring it later with webpack aliases? @maciejrybaniec @mczapkowicz @dariuszlacheta

File: src/components/FormatSettings/components/FormatTableSettings/constants.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  {
+    label: 'number',
+    value: DataTypes.number,
+  },
+  {
+    label: 'string',
+    value: DataTypes.string,
+  },
+  // {
```
Comment (apryka) [inline src/components/FormatSettings/components/FormatTableSettings/constants.ts:22]: do we need this? as well as `DataTypes.boolean`?

File: src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
+export const DATE_FORMATS = [
+  {
+    label: 'Original format',
```
Comment (apryka) [inline src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts:3]: should be wrapped with `t('...')`

File: src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  {
+    label: 'DD-MM-YYYY',
+    value: 'DD-MM-YYYY',
+  },
+  {
+    label: 'MM-DD-YYYY',
+    value: 'MM-DD-YYYY',
+  },
+  {
+    label: "Don't show date",
```
Comment (apryka) [inline src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts:19]: should be wrapped with `t('...')`

File: src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    value: 'hidden',
+  },
+];
+
+export const TIME_FORMATS = [
+  {
+    label: '12-hour clock',
```
Comment (apryka) [inline src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts:26]: should be wrapped with `t('...')`

File: src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    value: 'hh:mm',
+  },
+  {
+    label: '24-hour clock',
```
Comment (apryka) [inline src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts:30]: should be wrapped with `t('...')`

File: src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    value: 'HH:mm',
+  },
+  {
+    label: "Don't show time",
```
Comment (apryka) [inline src/components/FormatSettings/components/Formatters/DateTimeFormater/constants.ts:34]: should be wrapped with `t('...')`

File: src/components/FormatSettings/components/Formatters/NumericFormatter/NumericFormatter.tsx [hunk 13]
```diff
@@ -94,7 +109,17 @@ const FormatValues: FC<Props> = ({ formatValue, onUpdateFormatValue }) => {
         label: t(DEFAULT_FORMATTER_PATTERN.label),
         value: DEFAULT_FORMATTER_PATTERN.value,
       },
-      ...patterns,
+      ...patterns.map((pattern) => {
+        if (pattern.value === FULL_NUMBER_PATTERN) {
```
Comment (apryka) [inline src/components/FormatSettings/components/Formatters/NumericFormatter/NumericFormatter.tsx:113]: not necessary as `full number` should be removed. valid version of this part is on `develop` branch

File: src/components/ComponentSettings/components/Circular/components/ValueMode/ValueMode.tsx (hunk 6 above)
Comment (mczapkowicz) [reply src/components/ComponentSettings/components/Circular/components/ValueMode/ValueMode.tsx:5]: Yeah, webpack aliases definitely would be helpful :)

Comment (maciejrybaniec) [reply src/components/ComponentSettings/components/Circular/components/ValueMode/ValueMode.tsx:5]: Yes, @apryka could You please add ticket to our backlog - so we can discuss about it during refinement. 
If we are going to introduce Webpack aliases we need to think about them also in context of other front-end projects.

**Review by maciejrybaniec — APPROVED** [review]

**Review by apryka — APPROVED** [review]

**Pull request merged**